kernel/src/net/socket/ip/stream/listen.rs

```diff
@@ -57,7 +57,7 @@ impl ListenStream {
     }
 
     pub fn try_accept(&self) -> Result<ConnectedStream> {
-        let mut backlog_sockets = self.backlog_sockets.write();
+        let mut backlog_sockets = self.backlog_sockets.write_irq_disabled();
```

Comment:
Well, I admit that most of the lock usage in `net/socket/ip` is just wrong when it comes to IRQs. We're currently doing too many things in the IRQ handler, so there are a few more locks needed to disable IRQs.

But if you're going to change this, it might be better to at least change other `backlog_sockets` usages in the same file? Otherwise it looks pretty wired, as it's not at all clear what we want to do with this lock.